ansible/configs/ansible-middleware-workshop/files/cloud_providers/azure_cloud_template.j2

```diff
@@ -0,0 +1,428 @@
+            "version" : "latest"
+        },
+        "tenantId" : "[subscription().tenantId]",
+        "apiVersion" : "2015-06-15",
+        "apiVersionCompute" : "2015-06-15",
+        "apiVersionNetwork" : "2016-03-30",
+        "tmApiVersion" : "2015-11-01",
+        "apiVersionStorage" : "2015-06-15",
+        "apiVersionLinkTemplate" : "2015-01-01",
+        "nicName" : "OneVmNic",
+        "publicIPAddressType" : "Dynamic",
+        "subnetRef" : "[concat(variables('vnetID'),'/subnets/',variables('virtualNetworkName'))]",
+        "sshKeyPath" : "[concat('/home/',parameters('adminUsername'),'/.ssh/authorized_keys')]",
+        "sQuote" : "\"",
+        "vmStorageAccountContainerName": "vhds",
+        "storageAccountType": "Standard_LRS",
+        "vhdStorageType" : "Premium_LRS",
+        "storageAccountName": "[concat('vsts8',uniquestring(parameters('guid')))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Storage/storageAccounts",
+            "name": "[variables('StorageAccountName')]",
+            "apiVersion": "2016-01-01",
+            "location": "[resourceGroup().location]",
+            "sku": {
+                "name": "[variables('storageAccountType')]"
+            },
+            "kind": "Storage",
+            "properties": {
+            }
+        },
+{% for instance in instances %}
+
+{% if instance['dns_loadbalancer']|d(false)|bool and not instance['unique']|d(false)|bool %}
+        {
+            "type": "Microsoft.Network/dnszones/a",
+            "name": "[concat(variables('subzone'), '/', '{{instance['name']}}')]",
+            "apiVersion": "2016-04-01",
+            "dependsOn": [
+{% for c in range(1,(instance['count'] |int)+1) %}
+  {% if instance['unique']|d(false)|bool %}
+    {% set instancename = instance['name'] %}
+    {% else %}
+    {% set instancename = instance['name'] + (loop.index|string) %}
+  {% endif %}
+                "[resourceId('Microsoft.Network/publicIPAddresses/', '{{instancename}}-PublicIP')]",
+{% endfor %}
+                "[resourceId('Microsoft.Network/dnsZones/', variables('subzone'))]",
+            ],
+            "properties": {
+                "TTL": 3600,
+                "ARecords": [
+                    {
+{% for c in range(1,(instance['count'] |int)+1) %}
+  {% if instance['unique']|d(false)|bool %}
+    {% set instancename = instance['name'] %}
+    {% else %}
+    {% set instancename = instance['name'] + (loop.index|string) %}
+  {% endif %}
+                        "ipv4Address": "[reference('{{instancename}}-PublicIP').ipAddress]"
+{% endfor %}
+                    }
+                ]
+            }
+        },
+{% endif %}
+
+
+{% for c in range(1,(instance['count'] |int)+1) %}
+
+  {% if instance['unique']|d(false)|bool %}
+    {% set instancename = instance['name'] %}
+    {% else %}
+    {% set instancename = instance['name'] + (loop.index|string) %}
+  {% endif %}
+
+{% if instance['public_dns']|d(false)|bool %}
```

Comment:
This format will introduce some stray whitespace. I recommend aligning the `{%` brace at the first character in the line and putting the whitespace indentation within the braces:

```
{% for c in range(1,(instance['count'] |int)+1) %}
{%   if instance['unique']|d(false)|bool %}
{%     set instancename = instance['name'] %}
{%   else %}
{%     set instancename = instance['name'] + (loop.index|string) %}
{%   endif %}
{%   if instance['public_dns']|d(false)|bool %}
```